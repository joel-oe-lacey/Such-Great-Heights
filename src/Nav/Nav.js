import React from 'react';

const Nav = (props) => {
  return (
    <section className='nav-bar'>
      <img className='logo-nav'>
      <h2 className='username'>Welcome, [user]!</h2>
      <button className='view-areas-btn'>View Areas</button>
      <button className='log-out-btn'>Log Out</button>
    </section>
  )
}
