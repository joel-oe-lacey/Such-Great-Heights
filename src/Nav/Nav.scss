.nav-bar {
  font-family: 'Work Sans', sans-serif;
}
