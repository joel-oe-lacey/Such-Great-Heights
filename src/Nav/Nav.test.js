import React from 'react';
import { render } from '@testing-library/react';
import Nav from './Nav.js';
import { shallow } from 'enzyme';

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( <Nav /> );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot when there are props', () => {
    let props = {
      userName: 'user123'
    }
    wrapper = shallow(<Nav {...props} />)
    expect(wrapper).toMatchSnapshot();
  })


})
