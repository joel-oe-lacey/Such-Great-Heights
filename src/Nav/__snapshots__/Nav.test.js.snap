// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<section
  className="nav-bar"
>
  <img
    className="logo-nav"
  />
  <h2
    className="username"
  >
    welcome, undefined!
  </h2>
  <NavLink
    to="/favorites"
  >
    <button
      className="view-favorites-btn"
    >
      View Favorites
    </button>
  </NavLink>
  <NavLink
    to="/areas"
  >
    <button
      className="view-areas-btn"
    >
      View Areas
    </button>
  </NavLink>
  <NavLink
    to="/"
  >
    <button
      className="log-out-btn"
    >
      Log Out
    </button>
  </NavLink>
</section>
`;

exports[`Nav should match the snapshot when there are props 1`] = `
<section
  className="nav-bar"
>
  <img
    className="logo-nav"
  />
  <h2
    className="username"
  >
    welcome, user123!
  </h2>
  <NavLink
    to="/favorites"
  >
    <button
      className="view-favorites-btn"
    >
      View Favorites
    </button>
  </NavLink>
  <NavLink
    to="/areas"
  >
    <button
      className="view-areas-btn"
    >
      View Areas
    </button>
  </NavLink>
  <NavLink
    to="/"
  >
    <button
      className="log-out-btn"
    >
      Log Out
    </button>
  </NavLink>
</section>
`;
